compiler/src/dmd/lexer.d

```diff
@@ -701,12 +704,79 @@ class Lexer
                                 break IdentLoop;
                             }
 
+                            p += times - 1; // skip to last character in the hex value, incremented at top of loop
                             continue;
                         }
                         break;
                     }
 
-                    Identifier id = Identifier.idPool((cast(char*)t.ptr)[0 .. p - t.ptr], false);
+                    Identifier id;
+
+                    if (universalCharacterName4 == 0 && universalCharacterName8 == 0)
+                    {
+                        id = Identifier.idPool((cast(char*)t.ptr)[0 .. p - t.ptr], false);
+                    }
+                    else
+                    {
+                        auto priorValidation = t.ptr[0 .. p - t.ptr];
+                        const(char)* priorVPtr = priorValidation.ptr;
+                        stringbuffer.setsize(0);
+
+                        while(priorVPtr < &priorValidation[$-1])
```

Comment:
I think either the slice needs to be +1'd, or the condition be the end of the slice inclusive. Otherwise the last byte ends up being chopped off.
```suggestion
                        auto priorValidation = t.ptr[0 .. p - t.ptr + 1];
                        const(char)* priorVPtr = priorValidation.ptr;
                        stringbuffer.setsize(0);

                        while(priorVPtr < &priorValidation[$-1])
```
or
```suggestion
                        auto priorValidation = t.ptr[0 .. p - t.ptr];
                        const(char)* priorVPtr = priorValidation.ptr;
                        stringbuffer.setsize(0);

                        while(priorVPtr <= &priorValidation[$-1])
```